contracts/marketplace/lib.rs

```diff
@@ -0,0 +1,299 @@
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
I'd leave a crate-level comment here explaining what the contract it is, what are the mechanics etc.